cpp/src/search/contains_table.cu

```diff
@@ -134,22 +124,59 @@ std::pair<rmm::device_buffer, bitmask_type const*> build_row_bitmask(table_view
 }
 
 /**
- * @brief Invoke an `operator()` template with a row equality comparator based on the specified
- * `compare_nans` parameter.
+ * @brief Invokes the given `func` with desired comparators based on the specified `compare_nans`
+ * parameter
+ *
+ * @tparam HaystackHasNested Flag indicating whether there are nested columns in haystack
+ * @tparam HasAnyNested Flag indicating whether there are nested columns in either haystack or
```

Comment:
Why do we need two separate template parameters for "haystack has nested" and "haystack has nested or needles has nested"? I would have expected this to only require one dispatch. Is there a meaningful benefit to the case where "haystack has nested and not needles has nested"?